src/icrar/leap-accelerate/cuda/device_vector.h

```diff
@@ -57,14 +57,30 @@ namespace cuda
 
     public:
 
+        /**
+         * @brief Copy Constructor
+         * 
+         * @param other
+         */
+        device_vector(device_vector&& other) noexcept
+            : m_count(other.m_count)
+            , m_buffer(other.m_buffer)
+        {
+            other.m_buffer = nullptr;
+            other.m_count = 0;
+        }
+
+        device_vector& operator=(device_vector&& other) noexcept = default;
```

Comment:
It doesn't make sense that the move ctor is manually written, but the move assignment operator is defaulted, which is the correct answer?

Reply:
g++ says the default move constructor here is ill-formed (probably due to the pointer) and I assumed the operator gets implemented using the constructor. With some initial testing it seems also that the default only gets evaluated when its called via code. I'll write it explicitly.

Mainly I want all device objects to be noncopyable at the moment as nested copies are a bug rather than intended so far

Reply:
> I assumed the operator gets implemented using the constructor

No, it defaults to move each member of the object from `other` to `this`. So yes, it looks like you need to manually implement the move assignment operator to match the behavior of the move constructor.

In principle it would be better to use a `unique_ptr` to store your buffer (you get move methods for free, plus deleted copy methods), but that in turn would require you to provide a proper `Deleter` to the `unique_ptr` that internally does the call to `cudaFree`. Maybe not a bad idea for the future.